pkg/webhook/resources/virtualmachinebackup/mutator.go

```diff
@@ -0,0 +1,53 @@
+package virtualmachinebackup
+
+import (
+	"fmt"
+
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	ctlharvesterv1 "github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	werror "github.com/harvester/harvester/pkg/webhook/error"
+	"github.com/harvester/harvester/pkg/webhook/types"
+)
+
+type BackupMutator struct {
+	types.DefaultMutator
+	vmBackupCache ctlharvesterv1.VirtualMachineBackupCache
+}
+
+func NewMutator(vmBackupCache ctlharvesterv1.VirtualMachineBackupCache) *BackupMutator {
+	return &BackupMutator{vmBackupCache: vmBackupCache}
+}
+
+func (m *BackupMutator) Resource() types.Resource {
+	return types.Resource{
+		Names:      []string{"virtualmachinebackups"},
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   v1beta1.SchemeGroupVersion.Group,
+		APIVersion: v1beta1.SchemeGroupVersion.Version,
+		ObjectType: &v1beta1.VirtualMachineBackup{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+		},
+	}
+}
+
+func (m *BackupMutator) Create(_ *types.Request, newObj runtime.Object) (types.PatchOps, error) {
```

Comment:
How about handling the backup types individually? Say there is a snapshot with an identical name when the user creates a backup. And vice versa.

Reply:
Updated it. Thanks for the suggestion.